### PR TITLE
Add support for MXRT600

### PR DESCRIPTION
**Pull request by mmahadevan108**
The code currently can successfully run samples/hello_world & samples/basic/blinky

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @MaureenHelm

**Comment by mmahadevan108** [comment]
> So one change instead of mcux/devices/MIMXRT685S can this be more of a family so one could support both MIMXRT685S and MIMXRT633S w/o having to do a full import of all the driver files.

All drivers files are kept out of the device folder and can be shared between MIMXRT685S & MIMXRT633S. When adding MIMXRT633S, we should be able add a folder for MIMXRT633S under devices.

**Comment by galak** [comment]
> All drivers files are kept out of the device folder and can be shared between MIMXRT685S & MIMXRT633S. When adding MIMXRT633S, we should be able add a folder for MIMXRT633S under devices.

My point is that duplicating this for every chip is not scalable.  Thus suggesting that the dir with actual driver code be for the RT6xx family.

**Comment by mmahadevan108** [comment]
> > All drivers files are kept out of the device folder and can be shared between MIMXRT685S & MIMXRT633S. When adding MIMXRT633S, we should be able add a folder for MIMXRT633S under devices.
> 
> My point is that duplicating this for every chip is not scalable. Thus suggesting that the dir with actual driver code be for the RT6xx family.
The directory drivers/imxrtxxx is meant for the all RT6XX family, I can rename this to be more clear i.e drivers/imxrt6xx. 
We are in the process of restructuring the drivers folder under mcux to avoid such duplication.

**Comment by galak** [comment]
I'd suggest re-ordering the commits so that the file import is first, than the cmake additions.  Also, squash the removal of the power libs into the file import commit.

File: mcux/drivers/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #
-# Copyright (c) 2018, NXP
+# Copyright (c) 2018-2020, NXP
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
-if("${CONFIG_SOC_FAMILY}" STREQUAL "nxp_imx")
+if("${CONFIG_SOC}" STREQUAL "mimxrt685s_cm33")
```
Comment (galak) [inline mcux/drivers/CMakeLists.txt:7]: Can we change this from CONFIG_SOC to based on something like SOC_SERIES?

**Comment by galak** [comment]
Also rename the commit message for the file import

File: mcux/drivers/CMakeLists.txt (hunk 1 above)
Comment (mmahadevan108) [reply mcux/drivers/CMakeLists.txt]: Done

**Comment by mmahadevan108** [comment]
> I'd suggest re-ordering the commits so that the file import is first, than the cmake additions. Also, squash the removal of the power libs into the file import commit.

Done

**Comment by mmahadevan108** [comment]
> Also rename the commit message for the file import

Done

**Comment by mmahadevan108** [comment]
> LGTM. Probably good to update mcux/README

Done.

**Comment by galak** [comment]
looks like you need a rebase to deal with merge conflict on mcux/README.

**Comment by mmahadevan108** [comment]
Rebase done

**Pull request merged**